scripts/benchmark_all.sh

```diff
@@ -12,7 +12,17 @@ then
 fi
 
 benchmarks='hyriseBenchmarkTPCH hyriseBenchmarkTPCDS hyriseBenchmarkTPCC hyriseBenchmarkJoinOrder'
-num_mt_clients=50
+warmup_seconds=2
+runs=100
+
+# Setting the number of clients used for the multi-threaded scenario to the machine's physical core count.
+# This only works for macOS and Linux. For macOS, we simply divide the CPU count by 2.
+output="$(uname -s)"
+case "${output}" in
+    Linux*)     num_phy_cores="$(lscpu -p | egrep -v '^#' | grep '^[0-9]*,[0-9]*,0,0' | sort -u -t, -k 2,4 | wc -l)";;
+    Darwin*)    num_mt_clients="$(($(sysctl -n hw.ncpu) / 2))";;
```

Comment:
Is that "double subshell" thing done on purpose?

Reply:
I couldn't get it to work without it. The outer `(())` is needed because apparently this triggers _arithmetic expansion_.

Neither `Darwin*)    num_mt_clients="$((sysctl -n hw.ncpu / 2))";;` nor `Darwin*)    num_mt_clients="$(((sysctl -n hw.ncpu) / 2))";;` does work.